starlark/value.go

```diff
@@ -131,15 +131,36 @@ type Comparable interface {
 	CompareSameType(op syntax.Token, y Value, depth int) (bool, error)
 }
 
+// Alternate, simplified comparison interface
+type ThreeWayComparable interface {
```

Comment:
What do think of naming this TotallyOrdered.Cmp after the Python three-valued comparison function, cmp?

Let's document it as follows:

// A TotallyOrder is a type whose values form a total order:
// if x and y are of the same TotallyOrdered type, then x must be less than y,
// greater than y, or equal to y.
//
// It is simpler than Comparable and should be preferred in new code,
// but if a type implements both interfaces, Comparable takes precedence.
type ...